examples/plot_dataset_example.py

```diff
@@ -1,58 +1,133 @@
+    drop_samples=False)
+
+###############################################################################
+# We can iterate through the windows_ds which yields a supercrop/window x,
+# a target y, and supercrop_ind (which itself contains `i_supercrop_in_trial`,
+# `i_start_in_trial`, and `i_stop_in_trial`, which are required for combining
+# supercrop/window predictions in the scorer).
+for x, y, supercrop_ind in windows_ds:
+    print(x.shape, y, supercrop_ind)
     break
 
-# create a dataset based on TUH Abnormal EEG Corpus (v2.0.0)
-# for this dataset, no events exist but a label (pathological / non-pathological
-# is valid for the entire recording
-# ds = TUHAbnormal(path="/path/to/the/directory/",
-#                  subject_ids=[0, 1], trial_start_offset_samples=0,
-#                  trial_stop_offset_samples=1000, supercrop_size_samples=1000,
-#                  supercrop_stride_samples=1000, mapping={False: 0, True: 1})
-
-# as before, we can iterate through the dataset, getting the same kind of info
-# for x, y, info in ds:
-#     print(x.get_data().shape, y, info)
-#     break
-
-# we can change the target for this dataset to 'age'
-# ds = TUHAbnormal(path="/path/to/the/directory/",
-#                  subject_ids=[0, 1], trial_start_offset_samples=0,
-#                  trial_stop_offset_samples=1000, supercrop_size_samples=1000,
-#                  supercrop_stride_samples=1000, target="age",
-#                  mapping={False: 0, True: 1})
+###############################################################################
+# We visually inspect the supercrops/windows:
+max_i = 2
+fig, ax_arr = plt.subplots(1, max_i + 1, figsize=((max_i + 1) * 7, 5),
+                           sharex=True, sharey=True)
+for i, (x, y, supercrop_ind) in enumerate(windows_ds):
+    ax_arr[i].plot(x.T)
+    ax_arr[i].set_ylim(-0.0002, 0.0002)
+    ax_arr[i].set_title(f"label={y}")
+    if i == max_i:
+        break
+
+###############################################################################
+# Alternatively, we can create evenly spaced ("sliding") windows using a
+# different windower.
+sliding_windows_ds = create_fixed_length_windows(
+    ds, start_offset_samples=0, stop_offset_samples=None,
+    supercrop_size_samples=1200, supercrop_stride_samples=1000,
+    drop_samples=False)
+
+print(len(sliding_windows_ds))
+for x, y, supercrop_ind in sliding_windows_ds:
+    print(x.shape, y, supercrop_ind)
+    break
+
+###############################################################################
+# Transforms can also be applied on supercrops/windows in the same way as shown
+# above on continuous data:
+
+def crop_windows(windows, start_offset_samples, stop_offset_samples):
+    fs = windows.info["sfreq"]
+    windows.crop(tmin=start_offset_samples / fs, tmax=stop_offset_samples / fs,
+                 include_tmax=False)
+
+epochs_transform_dict = OrderedDict({
+    "pick_types": {"eeg": True, "meg": False, "stim": False},
```

Comment:
Lukas and I thought about discarding the stimulus channel right after getting the events, to prevent eventual training on data still containing the labels in the data array. That way you wouldn't need to call pick_types here again.

Reply:
To elaborate a bit more. For the BCIC IV 2a dataset, the events are defined by the stimulus channel. We call mne.find_events inside the create_windows_from_events function. This means that we still need the stimulus channel after preprocessing the raw. Additionally, we have to make sure that the user does not use the stimulus channel for decoding, since it contaisn the targets.

Reply:
I see, I think that makes sense. I suppose you would add that in the windower functions?

Also, I just realized `pick_types` doesn't work inplace (according to the documentation). If that's the case, or generally speaking for methods that don't work inplace, did you guys think about how that could be done? Would we have to define a wrapper around the method?

Reply:
> To elaborate a bit more. For the BCIC IV 2a dataset, the events are defined by the stimulus channel. We call mne.find_events inside the create_windows_from_events function. This means that we still need the stimulus channel after preprocessing the raw. Additionally, we have to make sure that the user does not use the stimulus channel for decoding, since it contaisn the targets.

One way around this would be to use MNE's annotations objects instead of stim channels. I know it adds some complexity to the code (especially since in BCI-like datasets it's convenient to work in samples instead of in seconds), but it's more natural for sleep recordings for instance.

Reply:
Yes, dropping the stimulus channel right after extracting the events in the windowing functions could be one way to do it.

I wasn't aware that pick_types does not work in place... Is there a way around it using mne? Maybe defining bad channels instead of picking channels?

Reply:
> One way around this would be to use MNE's annotations objects instead of stim channels. I know it adds some complexity to the code (especially since in BCI-like datasets it's convenient to work in samples instead of in seconds), but it's more natural for sleep recordings for instance.

So instead of using the stimulus channel in the windowing functions, you suggest that we use annotations, is that correct? And this means that we extract events from the stimulus channel in a preprocessing step and add them to the annotations before dropping channels and before windowing? Did I get this right?

Reply:
> So instead of using the stimulus channel in the windowing functions, you suggest that we use annotations, is that correct? And this means that we extract events from the stimulus channel in a preprocessing step and add them to the annotations before dropping channels and before windowing? Did I get this right?

Yes, that's it. The windowers would need to rely on mne.Annotations. I think the Dataset class itself (e.g., MOABBDataset) should create and populate an Annotations object to be placed in the Raw object (and then drop the stim channels).

Reply:
Especially as some MOABB datasets already contain the annotations object instead of a stim channel, that's a good thing to do, I think. MOABBDataset could just check if the annotations object is empty and populate it from the stim channel if that's the case. I think we implemented something similar in an earlier version already.

Reply:
Yes, you are right @sbbrandt. That's why we had (and still have but don't use) https://github.com/braindecode/braindecode/blob/4613b639976b6330c16ced433fd46a72c5cfdd08/braindecode/datasets/datasets.py#L21-L131.
It would be great to find a consistent way for all (MOABB) datasets, such that the windowing function will always get the events from mne.Annotations.